[Poh Yu Jie] iP

PokezardVGC opened the pull request:
# 1 Use GFMD in the PR description
**CS2103 best mod**
- [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/) (and Markdown in general) is useful in many places when using GitHub e.g., issue tracker, PR reviews, writing documentation. The aim of this task is to ensure that you are sufficiently familiar with the GFMD syntax.
- Requirements: Update (how?) the description of the iP PR you created earlier (do not add a new comment) so that it contains the following GFMD elements:
    1. a header
    2. a bullet list
    3. a numbered list
    4. a fenced code block (with syntax highlighting)
    5. a task list
    6. an emoji
    7. a blockquote
    8. a hyper link
    9. inline code
    10. some text formatting: bold, italic, stikethrough etc
```
System.out.println("Hello World");
```

-[x] create ShadowJar file
-[ ] finish up admin tasks

emoji trial 👍 

> this text is a quote

This is the [pokemon website](https://www.pokemon.com/us/)

Commit:
Add Gradle support

PokezardVGC committed:
Level 1

PokezardVGC committed:
level 2

PokezardVGC committed:
Level 3 without inheritances

PokezardVGC committed:
level 3 with classes

PokezardVGC committed:
Level 4 done

PokezardVGC committed:
Level 5

PokezardVGC committed:
Level 6 done, but change array to arraylist, implement collection

PokezardVGC committed:
Updated to arraylist

PokezardVGC committed:
fixed minor bugs, made code cleaner

PokezardVGC committed:
remained main to Duke

PokezardVGC committed:
no message

PokezardVGC committed:
uncommented try block

PokezardVGC committed:
attempted to find filepath

PokezardVGC committed:
Dat time, didn't add comparsion method

PokezardVGC committed:
add option to list activties for the day
format time

PokezardVGC committed:
error in creating files

PokezardVGC committed:
fixed

PokezardVGC committed:
removed redudant functions

PokezardVGC committed:
Merge branch 'branch-Level-7'

PokezardVGC committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

tinenhao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+import java.io.PrintWriter;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.text.SimpleDateFormat;
```
Consider including a line spacing between different import groups to make it look clearer

tinenhao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                String[] fromText = data.split(" \\| "); //special chara
+                boolean done = fromText[1].equals("1") ? true : false;
```
Consider changing the name of boolean to sound like a boolean

tinenhao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+        String command = sc.nextLine();
+        while (!command.split(" ")[0].equals("bye")) {
+            switch (command.split(" ")[0]) {
+                case "list":
```
Should remove the additional indentation of all the case lines

lfrostbytee commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+    }
+    @Override
+    public String toString() {
+        if(this.getDone()) {
```
Do remember to add a space between the if keyword and predicate.
```suggestion
        if (this.getDone()) {
```

lfrostbytee commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+    public String toString() {
+        if(this.getDone()) {
+            return (String.format("[D][X] %s (%s)", this.getVal(), this.getTiming()));
+        }
+        else {
```
Looks slightly jarring. Perhaps the [format](https://se-education.org/guides/conventions/java/intermediate.html#layout) specified by the Java Coding Standard can help make your if-else condition look neater.
```suggestion
        } else {
```

lfrostbytee commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+        return dateTime;
+    }
+
+    public boolean sameDay(LocalDateTime date) {
```
I like how you started the name of previous getter methods with _getSomething_. I think the format you used previously can be applied onto this method as well.

lfrostbytee commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+import java.io.PrintWriter;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.text.SimpleDateFormat;
 public class Duke {
```
Perhaps an additional empty line can help make your code cleaner?

```suggestion
import java.text.SimpleDateFormat;

public class Duke {
```

lfrostbytee reviewed (commented):
LGTM, almost. Just a few coding style errors to fix.

gqsg commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+
+public class Deadline extends Task {
+    private final LocalDateTime deadline;
+
```
Consider writing header comments for these public methods.

gqsg commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public static void createFiles() throws IOException {
```
Consider put this createFiles() function into another class that deals with the loading and saving the data, for example, you can create a storage class to implement all the functions that load and save the tasks into the file.

gqsg commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,67 @@
+        super(input, done, "");
+        this.timing = timing;
+    }
+    @Override
```
Consider leaving a space between the Event method and the Override keyword, this makes the code look cleaner

gqsg commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,49 @@
+        return new Task(this.val, false, "");
+    }
+
+    public boolean getDone() {
```
Consider named the boolean method to sound like boolean, for example, you can change it to isDone.

PokezardVGC committed:
Made it oop

PokezardVGC committed:
made it more oop

PokezardVGC committed:
packaged

PokezardVGC committed:
packaged

PokezardVGC committed:
no message

PokezardVGC committed:
gradle added

PokezardVGC committed:
gradle added

PokezardVGC committed:
tried to gradle

PokezardVGC committed:
no message

PokezardVGC committed:
added JavaDocs

PokezardVGC committed:
no message

PokezardVGC committed:
renamed to coding standards

PokezardVGC committed:
added function but gradle build in the way

PokezardVGC committed:
Deleted gradle files

PokezardVGC committed:
Merge gradle support branch

PokezardVGC committed:
gradle working

PokezardVGC committed:
nth to commir

PokezardVGC committed:
commit random lolz

PokezardVGC committed:
Gradle removed

PokezardVGC committed:
del gradle

PokezardVGC committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	.gradle/7.4/executionHistory/executionHistory.bin
#	.gradle/7.4/executionHistory/executionHistory.lock
#	.gradle/7.4/fileHashes/fileHashes.bin
#	.gradle/7.4/fileHashes/fileHashes.lock
#	.gradle/buildOutputCleanup/buildOutputCleanup.lock
#	.gradle/buildOutputCleanup/cache.properties
#	.gradle/buildOutputCleanup/outputFiles.bin
#	.idea/misc.xml
#	.idea/modules/Gradle/Lab_2.Gradle.iml
#	.idea/modules/Gradle/Lab_2.Gradle.main.iml
#	.idea/modules/Gradle/Lab_2.Gradle.test.iml
#	.idea/modules/Lab_2.iml
#	.idea/workspace.xml
#	src/test/java/ParserTest.java

PokezardVGC committed:
Revert "Merge branch 'A-JavaDoc'"

This reverts commit 1739300e8e172e0d2df03a798aeee300a88c30b3, reversing
changes made to b3646c8c8266662b49a13d6fc5a08ae48a25a2d7.

PokezardVGC committed:
deleted gradle

PokezardVGC committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	.gradle/7.4/executionHistory/executionHistory.bin
#	.gradle/7.4/executionHistory/executionHistory.lock
#	.gradle/7.4/fileHashes/fileHashes.bin
#	.gradle/7.4/fileHashes/fileHashes.lock
#	.gradle/buildOutputCleanup/buildOutputCleanup.lock
#	.gradle/buildOutputCleanup/cache.properties
#	.gradle/buildOutputCleanup/outputFiles.bin
#	.idea/gradle.xml
#	.idea/misc.xml
#	.idea/workspace.xml

PokezardVGC committed:
Merge branch 'A-CodingStandard' into HEAD

# Conflicts:
#	.gradle/7.4/checksums/checksums.lock
#	.gradle/7.4/dependencies-accessors/dependencies-accessors.lock
#	.gradle/7.4/executionHistory/executionHistory.bin
#	.gradle/7.4/executionHistory/executionHistory.lock
#	.gradle/7.4/fileHashes/fileHashes.bin
#	.gradle/7.4/fileHashes/fileHashes.lock
#	.gradle/buildOutputCleanup/buildOutputCleanup.lock
#	.gradle/buildOutputCleanup/cache.properties
#	.gradle/buildOutputCleanup/outputFiles.bin
#	.idea/gradle.xml
#	.idea/misc.xml
#	.idea/modules.xml
#	.idea/modules/Gradle/Lab_2.Gradle.iml
#	.idea/modules/Gradle/Lab_2.Gradle.main.iml
#	.idea/modules/Gradle/Lab_2.Gradle.test.iml
#	.idea/modules/Lab_2.iml
#	.idea/modules/Lab_2.test.iml
#	.idea/workspace.xml
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	src/main/java/duke/tasks/Task.java
#	src/main/java/duke/ui/Ui.java

PokezardVGC committed:
deleted all

PokezardVGC committed:
Merge branch 'actual_master_1'

# Conflicts:
#	.gradle/7.4/executionHistory/executionHistory.bin
#	.gradle/7.4/executionHistory/executionHistory.lock
#	.gradle/7.4/fileHashes/fileHashes.bin
#	.gradle/7.4/fileHashes/fileHashes.lock
#	.gradle/7.4/fileHashes/resourceHashesCache.bin
#	.gradle/buildOutputCleanup/buildOutputCleanup.lock
#	.gradle/file-system.probe
#	build/classes/java/main/duke/Duke.class
#	build/classes/java/main/duke/storage/Storage.class
#	build/classes/java/main/duke/taskList/TaskList.class
#	build/classes/java/main/duke/ui/Ui.class
#	build/tmp/compileJava/previous-compilation-data.bin
#	build/tmp/compileTestJava/previous-compilation-data.bin
#	src/main/java/duke/Duke.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/taskList/TaskList.java
#	src/main/java/duke/ui/Ui.java

PokezardVGC committed:
no message

PokezardVGC committed:
finalled merged

PokezardVGC committed:
merged

PokezardVGC committed:
Merge branch 'Level-9'

PokezardVGC committed:
restored files

PokezardVGC committed:
working

PokezardVGC committed:
realised not reading file

PokezardVGC committed:
added find function, gotten lost in git originally

PokezardVGC committed:
Merge tag 'Level-9'

PokezardVGC committed:
A-JUnit test onTaskList

PokezardVGC committed:
Task List test cases

PokezardVGC committed:
A-Jar

PokezardVGC committed:
current file

PokezardVGC committed:
changed all print in for UI to return Strings, and to be printed in Duke

PokezardVGC committed:
Ran with Launcher class but not reading inputs? need figure out how this works 1st

PokezardVGC committed:
javaFX installed

PokezardVGC committed:
UI prints outputs directly instead of as Strings

JavaFX needs to take in String inputs to run on GUI Interface

Changed all the print statements to return String function. Originally tested with print statements in Duke.java but later replaced to JavaFX methods

To allow statements to be printed on JavaFX GUI cleanly

{any other relevant info}

PokezardVGC committed:
Addition of Assert statements to TaskList class

Ensures that inputs of Mark, Unmark and Delete statements end with an integer so that we can gain access the the right position in TaskList

PokezardVGC committed:
Refactor methods such as toString in Deadline, Event, Task and ToDo classes

Enhance readability of code via refactoring

by converting long Strings or logic into variables

PokezardVGC committed:
Merge tag 'A-CodeQuality'

PokezardVGC committed:
Merged A-CodeQuality with A-Assertions

PokezardVGC committed:
Merge tag 'A-Assertions'

# Conflicts:
#	.idea/workspace.xml

Cleared merge conflicts

PokezardVGC committed:
Merge tag 'A-Assertions'

# Conflicts:
#	.idea/workspace.xml

PokezardVGC committed:
adds UpdateTime and UpdateTask features

Users to write in format of
updateTime 2 /to 26/11/2022 0800
updateTask 1 /to borrow another book

PokezardVGC committed:
Adds Ui.png

Shows example of interface

PokezardVGC committed:
adds ShadowJar

PokezardVGC commented:
# 1 Use GFMD in the PR description
**CS2103 best mod**
- [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/) (and Markdown in general) is useful in many places when using GitHub e.g., issue tracker, PR reviews, writing documentation. The aim of this task is to ensure that you are sufficiently familiar with the GFMD syntax.
- Requirements: Update (how?) the description of the iP PR you created earlier (do not add a new comment) so that it contains the following GFMD elements:
    1. a header
    2. a bullet list
    3. a numbered list
    4. a fenced code block (with syntax highlighting)
    5. a task list
    6. an emoji
    7. a blockquote
    8. a hyper link
    9. inline code
    10. some text formatting: bold, italic, stikethrough etc
```
System.out.println("Hello World");
```

-[x] create ShadowJar file
-[ ] finish up admin tasks

emoji trial 👍 

> this text is a quote

This is the [pokemon website](https://www.pokemon.com/us/)

PokezardVGC committed:
no message

PokezardVGC committed:
Create README.md

PokezardVGC committed:
Update README.md

PokezardVGC committed:
Update README.md

PokezardVGC committed:
Update README.md

PokezardVGC committed:
Update README.md

PokezardVGC committed:
Update README.md

PokezardVGC committed:
added Read.md

PokezardVGC committed:
Read me gotten shifted to docs file

PokezardVGC committed:
adds shadow jar

PokezardVGC committed:
stuff

PokezardVGC committed:
reader working now, need get rid of random task created at index 0

PokezardVGC committed:
fixes bugs

PokezardVGC committed:
adds shadowJar
